src/saltext/github/modules/github.py

```diff
@@ -1886,4 +1897,318 @@ def _query(
+        salt myminion github.get_ruleset ruleset_id=1
+    """
+
+    kwargs = _param_dict(rule_id=True, **kwargs)
+    params = _check_params(profile, **kwargs)
+    action = _format_action(rule_id=True, **params)
+
+    try:
+        ret = _query(profile, action, header_dict=params["header_dict"])
+        if not ret.get("error"):
+            return ret["dict"]
+        else:
+            ret["comment"] = f"GitHub Response Status Code: {ret.get('error')}"
+            ret["result"] = False
+    except CommandExecutionError:
+        ret = {}
+        ret["result"] = False
+        ret["comment"] = "Error getting ruleset"
+
+    return ret
+
+
+def delete_ruleset(profile="github", **kwargs):
+    """
+    Delete ruleset
+
+    profile
+        The name of the profile configuration to use. Defaults to ``github``.
+    ruleset_type
+        The type of ruleset. 'repo' or 'org'.
+    owner
+        The account owner of the repository.
+    repo_name
+        The name of the repo.
+    ruleset_id
+        Id of the ruleset.
+    org_name
+        The name of the organization.
+    header_dict
+        Headers to pass to Github API.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion github.delete_ruleset
+        salt myminion github.delete_ruleset ruleset_id=1
+    """
+    kwargs = _param_dict(rule_id=True, **kwargs)
+    params = _check_params(profile, **kwargs)
+    action = _format_action(rule_id=True, **params)
+
+    try:
+        ret = _query(profile, action, method="DELETE", header_dict=params["header_dict"])
+        if not ret.get("error"):
+            ret["comment"] = f"ruleset {params['ruleset_id']} successfully deleted"
+        else:
+            ret["comment"] = f"GitHub Response Status Code: {ret.get('error')}"
+            ret["result"] = False
+    except CommandExecutionError:
+        ret = {}
+        ret["result"] = False
+        ret["comment"] = "Error deleting ruleset"
+    return ret
+
+
+def update_ruleset(profile="github", **kwargs):
+    """
+    Update ruleset
+
+    profile
+        The name of the profile configuration to use. Defaults to ``github``.
+    ruleset_type
+        The type of ruleset. 'repo' or 'org'.
+    ruleset_id
+        Id of the ruleset.
+    owner
+        The account owner of the repository.
+    repo_name
+        The name of the repo.
+    org_name
+        The name of the organization.
+    header_dict
+        headers to pass to Github API.
+    ruleset_params
+        number of returned results (max 100).
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion github.update_ruleset
+        salt myminion github.update_ruleset ruleset_id=1
+    """
+    kwargs = _param_dict(rule_params=True, rule_id=True, **kwargs)
+    params = _check_params(profile, **kwargs)
+    action = _format_action(rule_id=True, **params)
+
+    if not isinstance(params["ruleset_params"], dict):
+        raise TypeError("params need to be a dict")
```

Comment:
**issue (blocking):** we don't want to raise anything other than `CommandExecutionError` in execution modules. it'll be caught by the salt framework and shown in a reasonable way. other exceptions will puke a stack trace.